src/components/Sidebar/NewFileModal.tsx

```diff
@@ -57,6 +58,20 @@ const NewFileModal = ({
     onClose()
   }, [])
 
+  useEffect(() => {
+    const handleKeyDownEvent = (event: KeyboardEvent) => {
+      if (event.key == 'Enter') {
+        onNewFile()
+      }
+    }
+
+    document.addEventListener('keydown', handleKeyDownEvent)
```

Comment:
Although this method works, it calls useEffect everytime the `newFileName` state changes. To avoid this, you could probably have a `onKeyDown` on the Input component and put this logic there. That would make things much cleaner!

Reply:
That seems way better. I'll update it today. Thanks!

Reply:
What's the consensus on separation of concerns? Because if we follow the concept of separation of concerns rigidly we would separate event handling and rendering entirely.

Reply:
Great question. Currently, my idea is to try and follow the best practices wherever possible. 
I have #23 opened up to refactor the entire code and separate business and rendering logic. I will work on that as soon as I resolve some other bugs and simple issues.

But meanwhile, if you have any good idea on how to achieve separation of concerns for this very specific case, I would be happy to know!

Reply:
If you have any ideas for refactoring, feel free to share in #23! 
Resolving this conversation...